internal/datacoord/services.go

```diff
@@ -1299,6 +1300,20 @@ func (s *Server) UpdateChannelCheckpoint(ctx context.Context, req *datapb.Update
 	}, nil
 }
 
+// ReportDataNodeTtMsgs send datenode timetick messages to dataCoord.
+func (s *Server) ReportDataNodeTtMsgs(ctx context.Context, req *datapb.ReportDataNodeTtMsgsRequest) (*commonpb.Status, error) {
+	if s.isClosed() {
+		log.Warn("failed to report dataNode ttMsgs on closed server")
+		return merr.Status(merr.WrapErrServiceUnavailable(msgDataCoordIsUnhealthy(s.session.ServerID))), nil
+	}
+
+	for _, ttMsg := range req.GetMsgs() {
+		go s.handleTimetickMessage(ctx, ttMsg)
```

Comment:
ttMsg arrived later might be handled earlier, this concurrency may cause one segment generating multiple Flush. `handleTimetickMessage` needs to be idempotent for a segment, and since we don't handle error here, might just log the error message in `handleTimetickMessage` rather than return it.